newrelic/structures_newrelic_synthetics_script_api_monitor.go

```diff
@@ -51,14 +61,24 @@ func buildSyntheticsScriptAPIMonitorUpdateInput(d *schema.ResourceData) syntheti
 	if v, ok := d.GetOk("locations_public"); ok {
 		input.Locations.Public = expandSyntheticsPublicLocations(v.(*schema.Set).List())
 	}
-	if v, ok := d.GetOk("script_language"); ok {
-		input.Runtime.ScriptLanguage = v.(string)
-	}
-	if v, ok := d.GetOk("runtime_type"); ok {
-		input.Runtime.RuntimeType = v.(string)
-	}
-	if v, ok := d.GetOk("runtime_type_version"); ok {
-		input.Runtime.RuntimeTypeVersion = synthetics.SemVer(v.(string))
+
+	_, scriptLangOk := d.GetOk("script_language")
+	_, runtimeTypeOk := d.GetOk("runtime_type")
+	_, runtimeVersionOk := d.GetOk("runtime_type_version")
+
+	if scriptLangOk || runtimeTypeOk || runtimeVersionOk {
```

Comment:
Any thoughts on changing to `and` as the API is expecting to have all the 3 values?

Reply:
Commented above :)